environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java

```diff
@@ -183,15 +188,15 @@ private EnvironmentDto updateAzureResourceEncryptionParameters(String accountId,
                 CreatedDiskEncryptionSet createdDiskEncryptionSet = environmentEncryptionService.createEncryptionResources(
                         environmentDtoConverter.environmentToDto(environment));
                 azureParameters.setDiskEncryptionSetId(createdDiskEncryptionSet.getDiskEncryptionSetId());
+                azureDiskEncryptionParametersRepository.save(azureParameters);
```

Comment:
This invocation - and the whole `AzureDiskEncryptionParametersRepository` actually - is not needed; the `environmentService.save(environment)` is expected to persist the changes to `azureParameters.diskEncryptionSetId` as well.

Reply:
`environmentService.save(environment)` stores the diskEncryptionSetId correctly. However, it does not persist the encryptionKeyUrl (which is a secret) - and is used during deletion of encryption resources. Hence, saving it using `AzureParametersRepository` now.